concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs

```diff
@@ -5,6 +5,7 @@ module Concordium.GlobalState.Persistent.BlockState.Updates where
 
 import Data.Foldable (toList)
 import qualified Data.Map as Map
+import qualified Data.Map.Strict as StrictMap
```

Comment:
I'd suggest changing to:
```suggestion
import qualified Data.Map.Strict as Map
```
All the existing uses of Map here may as well use the strict functions. (Note that `Data.Map.Strict` and `Data.Map` use the same underlying data structure. The only difference is that the functions in `Data.Map.Strict` force head-normal-form evaluation of values that are stored in the map.)